vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow/src/main/resources/META-INF/resources/frontend/menubarConnector.js

```diff
@@ -110,9 +116,11 @@ import './contextMenuConnector.js';
     };
   }
 
-  function setClassName (component) {
-    if (component._item) {
-      component._item.className = component.className;
+  function setClassName(component) {
+    const item = component._rootItem || component._item;
```

Comment:
It turns out `_item` still needs to be used for the submenu items (which doesn't get the `_rootItem` property from the `generateItems()` method.